cpp/arcticdb/version/version_map_batch_methods.hpp

```diff
@@ -17,6 +17,44 @@
 
 namespace arcticdb {
 
+struct SymbolStatus {
+    const VersionId version_id_ = 0;
+    const bool exists_ = false;
+    const timestamp timestamp_ = 0;
+
+    SymbolStatus(VersionId version_id, bool exists, timestamp ts) :
+        version_id_(version_id),
+        exists_(exists),
+        timestamp_(ts) {
+    }
+};
+
+inline std::shared_ptr<std::unordered_map<StreamId, SymbolStatus>> batch_check_latest_id_and_status(
+    const std::shared_ptr<Store> &store,
+    const std::shared_ptr<VersionMap> &version_map,
+    std::shared_ptr<std::vector<StreamId>> &symbols) {
+    ARCTICDB_SAMPLE(BatchGetLatestVersion, 0)
+    const LoadParameter load_param{LoadType::LOAD_LATEST_UNDELETED};
+    auto output = std::make_shared<std::unordered_map<StreamId, SymbolStatus>>();
+
+    async::submit_tasks_for_range(*symbols,
+        [store, version_map, &load_param](auto &symbol) {
+          return async::submit_io_task(CheckReloadTask{store, version_map, symbol, load_param});
+        },
+        [output](const auto& id, auto &&entry) {
+          auto index_key = entry->get_first_index(false);
+          if (index_key) {
+              output->insert(std::make_pair<StreamId, SymbolStatus>(StreamId{id}, {index_key->version_id(), true, index_key->creation_ts()}));
+          } else {
+              index_key = entry->get_first_index(true);
+              if (index_key)
+                  output->insert(std::make_pair<StreamId, SymbolStatus>(StreamId{id}, {index_key->version_id(), false, index_key->creation_ts()}));
```

Comment:
Is this correct? `entry->get_first_index(true)` will fetch the first index key in the version chain, but that isn't necessarily the "reference version id" we're looking for. e.g. if you do
`write -> write -> delete_version(1) -> delete_version(0)`
then `index_key->version_id()` here will be 0, but the reference version id is 1.